packages/data-stores/src/plans/queries/use-plans.ts

```diff
@@ -10,23 +9,23 @@ interface PlansIndex {
 }
 
 /**
- * Plans from `/plans` endpoint, indexed by `plan_slug`
+ * Plans from `/plans` endpoint, transformed into a map of planSlug => PlanNext
  */
-function usePlans() {
+function usePlans(): UseQueryResult< PlansIndex > {
 	const queryKeys = useQueryKeysFactory();
 
-	return useQuery< PricedAPIPlan[], Error, PlansIndex >( {
+	return useQuery( {
 		queryKey: queryKeys.plans(),
-		queryFn: async () =>
-			await wpcomRequest( {
+		queryFn: async () => {
```

Comment:
Same question for this new queryFn https://github.com/Automattic/wp-calypso/pull/85087/files#r1424662810